Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -900,11 +900,14 @@ Optional<HitTestResult> PaintableBox::hit_test(CSSPixelPoint position, HitTestTy
 {
     Optional<HitTestResult> result;
     (void)PaintableBox::hit_test(position, type, [&](HitTestResult candidate) {
-        VERIFY(!result.has_value());
-        if (!candidate.paintable->visible_for_hit_testing())
-            return TraversalDecision::Continue;
-        result = move(candidate);
-        return TraversalDecision::Break;
+        if (candidate.paintable->visible_for_hit_testing()) {
+            if (!result.has_value() || candidate.vertical_distance < result->vertical_distance || candidate.horizontal_distance < result->horizontal_distance)
```

Comment:
By checking the horizontal and vertical distances separately, you could end up with a point that is very close in one dimension but very far away in another. You'll probably get a better result by summing the horizontal and vertical distances before comparing

```suggestion
            if (!result.has_value() || candidate.horizontal_distance + candidate.vertical_distance < result->horizontal_distance + result->vertical_distance)
```

Reply:
The intention was to check them separately, I've attempted to make this a little clearer by making them optional. To show you can have one but not the other